src/app/modules/gb-side-panel-module/accordion-components/gb-queries/gb-queries.component.ts

```diff
@@ -35,12 +35,27 @@ export class GbQueriesComponent implements OnInit {
     uploadElm.click();
   }
 
+  processSubjectIdsUpload(fileContents: string, fileName: string): Query {
+    // we assume the text contains a list of subject Ids
+    let subjectIds: string[] = fileContents.split(/(\r\n)+/)
```

Comment:
maybe better to use `split(/\r?\n/)`? It seems like `split(/(\r\n)+/)` is not interpreting correctly new lines in Unix systems - I tested something like ` "  hello  \n\nfoo \n bar\nworld  \n"`, got `Array [ "hello  \n\nfoo \n bar\nworld" ]`

Reply:
I'll fix it

Reply:
And now conflicting changes....